.github/workflows/build.yml

```diff
@@ -42,22 +45,24 @@ jobs:
     steps:
       - name: Check out mitiq
         uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: "3.12"
+
       - name: Install Mitiq
         # Since requirements.txt includes cirq, which in turn has pyquil as dependency,
         # we explicitly remove pyquil from the installed packages after installing mitiq
-        run: |
-          python -m pip install --upgrade pip
-          pip install .
-          pip uninstall -y pyquil
```

Comment:
The "uninstall pyquil" is likely vestigial from when we transitioned from relying on `cirq` to `cirq-core`. Good to clean this up!